### ISSUE TITLE
Document the executor's default completion token

### ISSUE THREAD

**Issue by vinniefalco** [boostorg/beast#1750]
NT

**Comment by madmongo1** [comment]
The asio `use_awaitable.as_default_on()` example does not compile.

I have reported this issue in the Asio repo.
https://github.com/chriskohlhoff/asio/issues/477

Will revisit this issue in time for the next milestone.

**Comment by vinniefalco** [comment]
Yeah, I hate trying to get last-minute stuff in when it hasn't compiled for most of the development cycle. It can wait until 1.74, thanks.

**Comment by alandefreitas** [comment]
Where should this documentation go?

**Comment by vinniefalco** [comment]
Well for example `basic_stream` has a default executor:
https://www.boost.org/doc/libs/1_79_0/libs/beast/doc/html/beast/ref/boost__beast__basic_stream.html

And its async initiating functions have a default completion token:
https://www.boost.org/doc/libs/1_79_0/libs/beast/doc/html/beast/ref/boost__beast__basic_stream/async_read_some.html

We could first make sure that `as_default_on()` works (see Richard's comment above). And then maybe we could put an example code in the javadoc along with a test or two in the .cpp?
